### DISCUSSION TITLE
Spring Batch integration?

### DISCUSSION THREAD

**Discussion by bjansen** [beanio/beanio#135] [Ideas]
BeanIO 2.1 had a module to integrate it with Spring Batch.

In BeanIO 3, its author decided to drop the support for Spring Batch. I'm not sure why, maybe it was broken, maybe it was not used, maybe it was hard to maintain?

Anyway, I'm wondering if we should bring it back.

**Comment by jeffjensen** [comment]
Hi, thanks for assembling the various sources of BeanIO into this repo!  We are currently using BeanIO 2.1 and are rearchitecting some things, so we are wondering if can use BeanIO 3.x easily with Spring Batch or will we need to change libraries.  We have some manual assembly to do because of the 2.1 dropped classes; we'll start with those along with 3.x and see how it goes.

If this works for us, do you have interest in adding the classes (with any needed updates) back in?

**Reply by bjansen** [reply]
Hi,

BeanIO 3 is more or less the same as BeanIO 2.1, except the Maven coordinates changed. I haven't tested it with Spring Batch, but I don't think there's anything preventing the two from working together. You'll have to copy/paste [these 3 classes](https://github.com/gautamdudeja90/beanio/tree/master/src/org/beanio/spring) and possibly adapt them for newer versions of Spring Batch.

7 people upvoted this post, so I'll see how I can bring back an "official" Sprint Batch integration. Perhaps in a separate project, since BeanIO and Spring Batch have different release schedules.

**Reply by jeffjensen** [reply]
Hi, thanks for the reply.  Yes, great if you bring back the Spring Batch classes to BeanIO proper!

These are the initial changes we made to make the reader work, including for Java 21 (there's lots of further cleanup possible).  I committed the 2.x versions first, then applied our minimalist changes.
https://github.com/jeffjensen/beanio/commits/spring-batch

If I can help further, please let me know.

**Comment by davsclaus** [comment]
Please make this in a separate module like beanio-spring-batch or something as a lot of users are using beanio without having to pull in spring or other unwanted libraries.
